Fw/Types/StringUtils.cpp

```diff
@@ -29,3 +29,30 @@ U32 Fw::StringUtils::string_length(const CHAR* source, U32 max_len) {
     }
     return length;
 }
+
+FwSignedSizeType Fw::StringUtils::substring_find(const CHAR* source_string, FwSizeType source_size, const CHAR* sub_string, FwSizeType sub_size) {
+
+    FW_ASSERT(source_string != nullptr);
+    FW_ASSERT(sub_string != nullptr);
+
+    // zero size sub-strings should always match
+    if ((source_size > 0) && (0 == sub_size)) {
+        return 0;
+    }
+
+    for (FwSizeType source_index = 0; source_index < source_size; source_index++) {
+        // if the current character matches
+        for (FwSizeType sub_index = 0; sub_index < sub_size; sub_index++) {
+            // if there is a mismatch, go to next character
+            if (source_string[source_index + sub_index] != sub_string[sub_index]) {
+                break;
+            } else if (sub_index == sub_size - 1) {
```

Comment:
I feel like my inner @timcanham says "use `(sub_index == (sub_size - 1))` for clarity".